extensions/python/ExecutePythonProcessor.h

```diff
@@ -97,10 +99,8 @@ class ExecutePythonProcessor : public core::Processor {
     python_dynamic_ = true;
   }
 
-  void addProperty(const std::string &name, const std::string &description, const std::string &defaultvalue, bool required, bool el) {
-    python_properties_.emplace_back(
-        core::PropertyDefinitionBuilder<>::createProperty(name).withDefaultValue(defaultvalue).withDescription(description).isRequired(required).supportsExpressionLanguage(el).build());
-  }
+  void addProperty(const std::string &name, const std::string &description, const std::optional<std::string> &defaultvalue, bool required, bool el,
+      bool sensitive, const std::optional<int64_t>& property_type_code);
 
   const std::vector<core::Property> &getPythonProperties() const {
     return python_properties_;
```

Comment:
I think we should acquire the mutex here, too, and return a copy.  The copy will happen anyway at the calling site in `PythonCreator::registerScriptDescription`; locking the mutex adds some overhead, but it is necessary.

Reply:
Good point, updated in d3f47c57d21829d1715bd794f8c08a462f4fa361